furusiya/model/systems/system.py

```diff
@@ -0,0 +1,23 @@
+class System(type):
+    def __new__(mcs, clsname, bases, dct):
+        component_name = dct['_component_name']
+
+        def set_(cls, owner, component):
+            getattr(cls, f'{component_name}s')[owner] = component
```

Comment:
This voodoo needs a comment explaining what it's doing. Looks like it's looking for an attribute based on the component name, eg. `fighters` for `fighter` component. What's the `f` in front of the string?

Reply:
`Looks like it's looking for an attribute based on the component name, eg. fighters for fighter component.`
You got that right. Not only that, it also actually creates the attribute itself a little down the method.
I could've just made it a variable local to the `__new__` method, but you and I both know that's a very bad idea.

`What's the f in front of the string?`
See [f-strings](https://www.python.org/dev/peps/pep-0498/).